pkg/virt-controller/watch/vm.go

```diff
@@ -491,8 +491,14 @@ func (c *VMController) handleDataVolumes(vm *virtv1.VirtualMachine, dataVolumes
 			c.dataVolumeExpectations.ExpectCreations(vmKey, 1)
 			curDataVolume, err = c.clientset.CdiClient().CdiV1beta1().DataVolumes(vm.Namespace).Create(context.Background(), newDataVolume, v1.CreateOptions{})
 			if err != nil {
-				c.recorder.Eventf(vm, k8score.EventTypeWarning, FailedDataVolumeCreateReason, "Error creating DataVolume %s: %v", newDataVolume.Name, err)
 				c.dataVolumeExpectations.CreationObserved(vmKey)
+				if pvc != nil && strings.Contains(err.Error(), "already exists") {
```

Comment:
Can we use the function [IsAlreadyExists](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsAlreadyExists) instead of the string parsing

Reply:
No, this is a different error.  It is not returned by kube-apiserver rather it is returned by cdi-apiserver here:  https://github.com/kubevirt/containerized-data-importer/blob/main/pkg/apiserver/webhooks/datavolume-validate.go#L584-L589